static/app/components/smartSearchBar/index.tsx

```diff
@@ -314,9 +314,9 @@ class SmartSearchBar extends React.Component<Props, State> {
     const {query} = this.props;
     const {query: lastQuery} = prevProps;
 
-    if (query !== lastQuery && defined(query)) {
+    if (query !== lastQuery && (defined(query) || defined(lastQuery))) {
       // eslint-disable-next-line react/no-did-update-set-state
-      this.setState(makeQueryState(addSpace(query)));
+      this.setState(makeQueryState(addSpace(query ?? undefined)));
```

Comment:
Was this something typescript should have caught but didn't?

Reply:
In other words, are our types wrong?

Reply:
Considering that state is derived from props, this should be `static getDerivedStateFromProps(props, state)`.

Reply:
The problem was in the logic of the if statement, types are fine.